core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -490,23 +490,22 @@ class TimeSeriesShard(val dataset: Dataset,
   def partKeysWithFilters(filter: Seq[ColumnFilter],
                              endTime: Long,
                              startTime: Long,
-                             limit: Int): List[ZeroCopyUTF8String] = {
-    val result: Set[ZeroCopyUTF8String] = new mutable.HashSet[ZeroCopyUTF8String]()
+                             limit: Int): Iterator[TimeSeriesPartition] = {
+    val result = new ArrayBuffer[TimeSeriesPartition]()
```

Comment:
It is a waste to create an ArrayBuffer here, that uses up O(n) memory.  You already have an underlying partIterator, just transform it via map.  IE:

    partIterator.map(getPartitionFromPartId)

Much shorter, uses much less memory, and much easier on the eyes and easier to reason about. :)

Reply:
@velvia 👍, added an implicit for adding map feature to IntIterator